gammapy/detect/tests/test_cwt.py

```diff
@@ -1,12 +1,109 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 from __future__ import absolute_import, division, print_function, unicode_literals
-from ...utils.testing import requires_dependency
-from ...detect import CWT
+import numpy as np
+from astropy.io import fits
+from numpy.testing import assert_allclose
+from ...utils.testing import requires_dependency, requires_data
+from ...detect import CWT, CWTKernels, CWTData
+from ...datasets import load_poisson_stats_image
 
 
 @requires_dependency('scipy')
-def test_CWT():
-    cwt = CWT(nscales=6, min_scale=6.0, scale_step=1.3)
+@requires_data('gammapy-extra')
+class TestCWT:
+    """
+    Test CWT algorithm, CWTData and CWTKernels.
 
-    # TODO: run on test data
-    assert 42 == 42
+    TODO: add test for __sub__ in CWTData?
+    """
+
+    def setup(self):
+        filename = load_poisson_stats_image(return_filenames=True)
+        image = fits.getdata(filename, hdu='COUNTS')
+        background = fits.getdata(filename, hdu='BACKGROUND')
+        header = fits.getheader(filename, hdu='COUNTS')
+
+        self.kernels = CWTKernels(n_scale=2, min_scale=6.0, step_scale=1.3)
+        self.data = dict(image=image, background=background, header=header)
+        self.cwt = CWT(kernels=self.kernels, significance_threshold=70., keep_history=True)
+
+    def test_cwt_kernels_creation(self):
+        assert_allclose(self.kernels.scales, [6., 7.8])
```

Comment:
Before we continue with the CWT review, please debug the old / new kernels and add one test that asserts the kernels here.

Having two kernels that are almost the same scale (6, 7.8) is probably non-ideal, because CWT results obtained with them are so similar. Why not make the kernels more different in size, e.g. scale = 3 and 7.8?